src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.Helpers.cs

```diff
@@ -36,7 +36,7 @@ private static void GetRuntimePropertyInfo(object value, JsonClassInfo jsonClass
             }
         }
 
-        private static void VerifyValueAndType(object value, Type type)
+        private static void VerifyValueAndType(object? value, Type? type)
```

Comment:
I am fine with this particular method that checks for nullable type to remain nullable.